File: 7kyu/nextPerfectSquare/challenge.js
/**
 * Challenge: Find the next perfect square!
 */

function findNextSquare(sq) {
  // Return the next square if sq is a perfect square, -1 otherwise
  let root = Math.sqrt(sq);
  return Number.isInteger(root) ? Math.pow(root + 1, 2) : -1;
}

File: 7kyu/nextPerfectSquare/challenge.rs
/**
 * Challenge: Find the next perfect square!
 */

fn find_next_square(sq: u64) -> Option<u64> {
    let root = (sq as f64).sqrt();
    return if root.fract() == 0.0 {
        Some((root + 1.0).powi(2) as u64)
    } else {
        None
    };
}
